github/is-member-of/action.yml

```diff
@@ -24,13 +24,20 @@ runs:
     - id: gh-api-is-member
       name: Check if user is member of the given org
       run: |
-        if gh api \
+        # required because 'gh api' will return non-zero for 404s to indicate lack of membership
+        set +e
+        # gh api will return a 404 with status to indicate lack of membership
+        result=$(gh api \
           -H "Accept: application/vnd.github+json" \
-          /orgs/${{ env.GH_ORG }}/members/${{ env.GH_USER }} ; then
+          /orgs/${{ env.GH_ORG }}/members/${{ env.GH_USER }} 2>/dev/null | jq -r '.status // "member"' )
+
+        if [[ '404' != "${result}" ]]; then
           echo "result=true" >> $GITHUB_OUTPUT
         else
+          # 404 or unknown user type
           echo "result=false" >> $GITHUB_OUTPUT
         fi
+        set -e
```

Comment:
`nit`: I think we don't need this context switch, however I do like this practice of resetting to the previous state.

Reply:
I will reduce the scope by moving this before the `if` to make it more clear.